src/eos-updater-util.c

```diff
@@ -367,14 +367,18 @@ eos_updater_resolve_upgrade (EosUpdater  *updater,
   shuffle_out_values (upgrade_ref, p_ref, NULL);
```

Comment:
nit: Consider explaining in the third commit message why the synchronous API should be used here rather than the asynchronous API.